model/integration_action.go

```diff
@@ -204,6 +204,7 @@ type DialogElement struct {
 
 type OpenDialogRequest struct {
 	TriggerId string `json:"trigger_id"`
+	UserId    string `json:"user_id"`
```

Comment:
I'm 3/5 that this is not needed and we should not confuse "normal" integrations with this field.

When taking a broad look at this changes it seams to be unneeded to include this here. To me the plugin API method signature should be `(userid string, request model.OpenDialogRequest) *model.AppError`. What do you think about adding a new API method called e.g. `OpenInteractiveDialogWithoutTriggerId` and depreciate the old one? This seams to me the clean approach. @scottleedavis @jwilander Thoughts?

Reply:
I chose adding UserId to OpenDialogRequest to have as little impact as possible on the API definitions, however if it is preferred to have the new API method and the deprecated method, I'd be happy to do that.  Though I would prefer less of a lengthy name than `OpenInteractiveDialogWithoutTriggerId`.  Maybe `OpenInteractiveDialogAsUser` or... ?   I don't have a better recommendation currently.

Reply:
My idea is, to consolidate the two methods when the next version of the plugin API is released and only have a method that doesn't need a trigger.

My problem with `OpenInteractiveDialogAsUser` is that it suggests that `OpenInteractiveDialog` doesn't opens the dialog as a user.

@levb @cpoile would like to here your thoughts on this discussion if you have the time.